gammapy/modeling/tests/test_sampling.py

```diff
@@ -0,0 +1,44 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from gammapy.datasets import Datasets
+from gammapy.modeling.sampling import run_mcmc, ln_uniform_prior
+import numpy as np
+from gammapy.utils.testing import requires_dependency
+
+def get_datasets():
+    dataset = Datasets.read("$GAMMAPY_DATA/fermi-3fhl-crab/Fermi-LAT-3FHL_datasets.yaml",
+                            "$GAMMAPY_DATA/fermi-3fhl-crab/Fermi-LAT-3FHL_models.yaml")
+
+    # Define the free parameters and min, max values
+    parameters = dataset.models.parameters
+    parameters["lon_0"].frozen = False
+    parameters["lat_0"].frozen = False
+    parameters["norm"].frozen = True
+    parameters["alpha"].frozen = True
+    parameters["beta"].frozen = True
+    parameters["lat_0"].min = -90
+    parameters["lat_0"].max = 90
+    parameters["lon_0"].min = 0
+    parameters["lon_0"].max = 360
+    parameters["amplitude"].min = 0.01 * parameters["amplitude"].value
+    parameters["amplitude"].max = 100 * parameters["amplitude"].value
+
+    return dataset
+
+def test_lnprob():
+    #Testing priors
+    dataset = get_datasets()
+    parameters = dataset.models.parameters
+
+    #paramater is within min, max boundaries
+    assert ln_uniform_prior(dataset) == 0.0
+    # Setting amplitude outside min, max values
+    parameters["amplitude"].value = 1000
+    assert ln_uniform_prior(dataset) == -np.inf
+
+@requires_dependency("emcee")
+def test_runmcmc():
+    #Running a small MCMC on pregenerated datasets
+    dataset = get_datasets()
+    run_mcmc(dataset, nwalkers=6, nrun=10)  # to speedup the test
+
```

Comment:
There is nothing to assert here?

Reply:
I added assert that the sampler returned is an emcee.ensemble.EnsembleSampler  object.
As the walker are random hard to test any specific value.